asserts/snap_asserts_test.go

```diff
@@ -154,6 +157,44 @@ func (sds *snapDeclSuite) TestDecodeInvalid(c *C) {
 
 }
 
+func (sds *snapDeclSuite) TestDecodePlugs(c *C) {
+	encoded := `type: snap-declaration
+authority-id: canonical
+series: 16
+snap-id: snap-id-1
+snap-name: first
+publisher-id: dev-id1
+plugs:
+  interface1:
+    allow-auto-connection:
+      slot-attributes:
+        a1: /foo/.*
+  interface2:
+    allow-connection:
+      slot-attributes:
+        a2: /foo/.*
```

Comment:
Can we also add similar deny-connection, deny-auto-connection, allow-installation and deny-installation tests? Also have slot-publisher-id, slot-snap-id, slot-snap-type and plug-attributes represented here would be good (ie, having a valid example of each field in this toplevel snap declaration test is useful).